.eslintignore

```diff
@@ -2,6 +2,7 @@ core/templates/expressions/parser.js
 core/templates/google-analytics.initializer.ts
 backend_prod_files/*
 core/tests/protractor.conf.js
+core/tests/protractor-browserstack.conf.js
```

Comment:
Can you please explain why we have added this file here?

Reply:
All the files present in the codebase are either to be included in the project configuration or are to be added in the .eslintignore file, So that the parser is either able to go through them or ignore them completely.
The following error was arising if we provided the 'tsconfig.json' file to parser:
![Screenshot from 2021-04-20 15-08-07](https://user-images.githubusercontent.com/68547101/116842535-78b16f00-abfa-11eb-8426-2edcc2b74044.png)

Thus all these files were to included in the project or completely ignored. To make no change in the tsconfig.json, I created another file 'parser-tsconfig.eslint.json' which simply just extends are tsconfig.json by including all the other files such as core/tests and other excluded files.As adding them to .eslintignore

Reply:
I think the typescript-related lint check should sync with our typescript config/coverage in the codebase. 

1. Do you know why the typescript-compilation which uses 'tsconfig.json'  doesn't fail for those files?
2. Is it recommended (on ts-eslint parser) to have separate config file for linting and ts-compilation?
3. Can we do some changes in 'tsconfig.json' to fix the issue? (Assuming that it won't affect ts-compilation)

/cc @vojtechjelinek Do let us know if you have any thoughts?

Reply:
I think we can try to put all the stuff that is missing to the _tsconfig.json_ if it doesn't produce too many errors, if it produces too much I think we can create a new config, but we should make sure to somehow keep these two in sync. So for example the _tsconfig.json_ should always be a subset of _parser-tsconfig.eslint.json_.

Reply:
Thanks for your suggestion, @vojtechjelinek In that case I think we should only have a separate config file in the worst case (to avoid ensuring the two files are in sync and managing multiple files).

Reply:
@Rijuta-s Is that helpful? Do you have any questions, let me know I'll be happy to help and get this thread resolved quickly! :)

Reply:
@DubeySandeep @vojtechjelinek Will you please have a look at the failing tests (typescript and e2e). I have included all the files in the tsconfig.json file. I can try fixing all the typescript checks but I think that too will require some time.

Maybe we can merge this PR without the lint check and ill raise another PR with the lint check as there are too many merge conflicts and bad patterns coming in due to ongoing migration as they are creating files with old function names which I have changed in the files.

Reply:
As per offline discussion, we are going planning to add lint check in a separate PR! (@Rijuta-s Can you please create an issue for the same and assign it to yourself?)

@Rijuta-s Let's get this PR merged quickly as it changes 200+ files and seems prone to merge conflicts.